packages/components/src/utils/getTextAsHtml.ts

```diff
@@ -20,31 +22,79 @@ export const getTextAsHtml = (content?: (HardBreakProps | TextProps)[]) => {
     return ""
   }
 
-  return content
-    .map((node) => {
-      if (node.type === "hardBreak") {
-        return "<br />"
-      }
+  const output: string[] = []
+  let existingLinkMark: Marks | undefined = undefined
+
+  // At every step, we will close off all marks except for links
+  // First encounter with a link, always open it first before other marks
+  // Close all other marks first before closing the link mark
+  content.forEach((node) => {
+    if (node.type === "hardBreak") {
+      output.push("<br />")
+      return
+    }
+
+    const newLinkMark = node.marks?.find((mark) => mark.type === "link")
+    const isLinkMarkNew =
+      !!newLinkMark && !isEqual(existingLinkMark, newLinkMark)
+
+    // Close off the existing link mark if it is different
+    if (isLinkMarkNew && !!existingLinkMark) {
+      output.push(`</${MARK_DOM_MAPPING.link}>`)
+      existingLinkMark = undefined
+    }
 
-      if (!node.marks) {
-        return node.text
-      }
+    // If there are no marks, just push the text
+    if (!node.marks) {
+      output.push(node.text)
+      return
+    }
```

Comment:
nit: for readability reasons, could we put this condition further up

Reply:
Clarified offline, can't do that here because we need to make sure we close off any existing open link marks first before we can push new texts.